raiden_contracts/tests/test_channel_settle.py

```diff
@@ -0,0 +1,135 @@
+from raiden_contracts.utils.config import E_CHANNEL_SETTLED, SETTLE_TIMEOUT_MIN
+from raiden_contracts.utils.events import check_channel_settled
+from .utils import get_settlement_amounts
+from .fixtures.config import fake_hex
+
+
+def test_update_channel_state(
+        web3,
+        custom_token,
+        token_network,
+        create_channel,
+        channel_deposit,
+        get_accounts,
+        get_block,
+        create_balance_proof
+):
+    (A, B) = get_accounts(2)
+    settle_timeout = SETTLE_TIMEOUT_MIN
+    deposit_A = 20
+    additional_hash = fake_hex(32, '02')
+    locksroot1 = fake_hex(32, '03')
+    locksroot2 = fake_hex(32, '04')
+
+    # Create channel and deposit
+    channel_identifier = create_channel(A, B, settle_timeout)
+    channel_deposit(channel_identifier, A, deposit_A)
+
+    # Create balance proofs
+    balance_proof_A = create_balance_proof(
+        channel_identifier,
+        A, 10, 2, 5,
+        locksroot1, additional_hash
+    )
+    balance_proof_B = create_balance_proof(
+        channel_identifier,
+        B, 5, 1, 3,
+        locksroot2, additional_hash
+    )
+    balance_proof_BA = create_balance_proof(
+        channel_identifier,
+        B, 10, 2, 5,
+        locksroot1, additional_hash
+    )
+
+    # Close channel and update balance proofs
+    token_network.transact({'from': A}).closeChannel(*balance_proof_B)
+    token_network.transact({'from': B}).updateNonClosingBalanceProof(
+        *balance_proof_A,
+        balance_proof_BA[4]
+    )
+
+    # Settlement window must be over before settling the channel
+    web3.testing.mine(settle_timeout)
```

Comment:
I don't know how the testing mining is implemented. Is it real Proof of Work. Is it proof of Authority?

Looking at their code there also seems to be a ["time travel"](https://github.com/ethereum/web3.py/blob/master/web3/testing.py#L7) option.